nimare/nimads.py

```diff
@@ -0,0 +1,215 @@
+        ...
+
+    def get_analyses_by_coordinates(self, xyz, r=None, n=None):
+        """Extract a list of Analyses with at least one Point near the requested coordinates."""
+        ...
+
+    def get_analyses_by_mask(self, img):
+        """Extract a list of Analyses with at least one Point in the specified mask."""
+        ...
+
+    def get_analyses_by_annotations(self):
+        """Extract a list of Analyses with a given label/annotation."""
+        ...
+
+    def get_analyses_by_texts(self):
+        """Extract a list of Analyses with a given text."""
+        ...
+
+    def get_analyses_by_images(self):
+        """Extract a list of Analyses with a given image."""
+        ...
+
+    def get_analyses_by_metadata(self):
+        """Extract a list of Analyses with a metadata field/value."""
+        ...
+
+    def get_points(self, analyses):
+        """Collect Points associated with specified Analyses."""
+        ...
+
+    def get_annotations(self, analyses):
+        """Collect Annotations associated with specified Analyses."""
+        ...
+
+    def get_texts(self, analyses):
+        """Collect texts associated with specified Analyses."""
+        ...
+
+    def get_images(self, analyses):
+        """Collect image files associated with specified Analyses."""
+        ...
+
+    def get_metadata(self, analyses):
+        """Collect metadata associated with specified Analyses."""
+        ...
+
+
+class Study:
+    """A collection of Analyses from the same paper.
+
+    Attributes
+    ----------
+    id : str
+        A unique identifier for the Study.
+    analyses : list of Analysis objects
+        The Analysis objects comprising the Study.
+    """
+
+    def __init__(self):
+        ...
+
+    def get_analyses(self):
+        """Collect Analyses from the Study.
+
+        Notes
+        -----
+        What filters, if any, should we support in this method?
+        """
+        ...
+
+
+class Analysis:
+    """A single statistical analyses from a Study.
+
+    Attributes
+    ----------
+    id : str
+        A unique identifier for the Analysis.
+    conditions : list of Condition objects
+        The Conditions in the Analysis.
+    annotations : list of Annotation objects
+        Any Annotations available for the Analysis.
+        Each Annotation should come from the same Annotator.
+    texts : dict
+        A dictionary of source: text pairs.
+    images : dict of Image objects
+        A dictionary of type: Image pairs.
+    points : list of Point objects
+        Any significant Points from the Analysis.
+
+    Notes
+    -----
+    Should the images attribute be a list instead, if the Images contain type information?
+
+    Should the conditions be linked to the annotations, images, and points at all?
+    """
+
+    def __init__(self):
+        ...
+
+
+class Condition:
+    """A condition within an Analysis.
+
+    Attributes
+    ----------
+    name
+    description
```

Comment:
This would fit under annotations, current thinking is if there are multiple conditions each column in annotations could represent the condition-specific trial count (e.g., `switch_trial_count`, `repeat_trial_count`, etc.). if a meta-analysis needed to use those specific columns it would have to be specified during the fit process of the estimator with the dataset/annotations tied to the estimator. I don't know how that information would be used off the top my head, so I do not know how the relationship between the condition columns would need to be specified. How does that sound?

Reply:
Thanks for the clarification. I'll add a note to the docstring so it's clear what Condition-specific info is offloaded to Analysis-level Annotations.

I'm sure NiMARE will eventually be able to handle arbitrary weights from Annotations (e.g., #363), and trial count/sample size will probably be a pretty common weight, so I think that approach makes sense.